source/common/stats/stats_impl.h

```diff
@@ -241,11 +240,16 @@ struct RawStatData {
    */
   bool initialized() { return name_[0] != '\0'; }
 
+  /**
+   * Returns the name as a string_view with no truncation.
+   */
+  absl::string_view key() const { return absl::string_view(name_); }
+
   /**
    * Returns the name as a string_view. This is required by BlockMemoryHashSet.
    */
-  absl::string_view key() const {
-    return absl::string_view(name_, strnlen(name_, maxNameLength()));
+  absl::string_view keyGivenStatsOptions(const StatsOptions& stats_options) const {
+    return absl::string_view(name_, strnlen(name_, stats_options.maxNameLength()));
```

Comment:
I'm kinda on-board with this.  Can we not just pre-truncate the name?  As in, why can't we just pass an optional length to which we want `name_` truncated?  As @jmarantz mentioned, this will require us to make `name_` null-terminated.

Reply:
This gets truncated at write time in `initialize()` / `safeInitialize()`; i *think* we can remove the `keyGivenStatsOptions` entirely since `name_` is already being null-terminated.

Reply:
Yes, this works! I will replace all instances of `keyGivenStatsOptions` with `key()`.